scripts/apps/authoring/authoring/controllers/ChangeImageController.js

```diff
@@ -32,6 +32,17 @@ export function ChangeImageController($scope, gettext, notify, modal, _, api, $r
         isDirty: false,
     };
 
+    const EDITABLE_METADATA = [
```

Comment:
Great! Could you also add a comment including the name of the constant to `scripts/apps/authoring/media/views/media-metadata-editor-directive.html`?

Reply:
> The are a lot of directives that uses different fields

We'd have a bug if I added a new field to `media-metadata-editor-directive.html`, but not to the constants. I think that's related enough to add a comment. I'm not asking to refactor `media-metadata-editor-directive.html` to use a repeater over the constants, only to add a comment.

I can add it myself if you changed your mind regarding me pushing to your repository :)

Reply:
On all directives you have to keep controller and the view in synch when you change them.
In a way or other the name of the fields are hardcoded on both controller and view.
Again it is about to be pragmatic and spent time in what matters. 

>I can add it myself if you changed your mind regarding me pushing to your repository :)
It is a bad solution to solve opinion differences.

Reply:
Here's why I think it does matter. Take a look at this [scheme](https://user-images.githubusercontent.com/2076953/45873205-88a61880-bd9a-11e8-9184-889bbca2b9a3.png). If someone is working on "Controller 2" and needs to change "directive 2", it's not realistic that the person will read the code of all controllers which's views use "directive 2". Even if he does, he might not notice the dependency, that's why I think it's helpful to add a comment about it.

Reply:
I think the developer follow the action(function call) from view and always will find the right controller where the function is.
The call graph in superdesk is complicated in many cases and we didn't used any kind of documentation for this. 
I think this is a good subject to talk about on next sprint retrospective.

Reply:
> I think the developer follow the action(function call) from view and always will find the right controller where the function is.

I totally agree that if you see a function call in a directive, it's not hard to find the controller where the function is defined. The hard thing is checking all views using a directive and their corresponding controllers to find an _implicit_ dependency which is not obvious.

For example there is no easy way to find out without checking all related code, that if you add a field in say `media-metadata-editor-directive.html`, you should also add it to the constant you defined. The proper solution would be using a repeater in the directive instead of hardcoding the fields, but that might take more time than we would like, that's why I suggested the commenting "solution".

Yeah, we can talk about it in the retrospective :)